### ISSUE TITLE
linter: False positive in `prefer-promise-reject-errors` for narrowed variables

### ISSUE THREAD

**Issue by hesprs** [oxc-project/oxc#25232]
### What version of Oxlint are you using?

1.76.0

### What command did you run?

oxlint -A all -D prefer-promise-reject-errors repro.ts

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

No config file, reproduced minimally using the command above.

### What happened?

`repro.ts`:

```TypeScript
declare const error: Error | undefined;
new Promise((_, reject) => {
    if (error) reject(error);
});
```

Oxlint reports: `error eslint(prefer-promise-reject-errors): Expected the Promise rejection reason to be an Error.` But `error` is already narrowed to an error.

Eslint 10.8.0 does not report similar false positive.

Type-aware does not change the result.

**Comment by camc314** [comment]
Hi! Thanks for reporting this issue.

This is a bug in the eslint version, however the typescript eslint should work as expected.

> Type-aware does not change the result.
I checked the type aware version and it appears to work as expected here.

Note the typescript riule needs to be explicilty enabled:

```json
{
  "options": { "typeAware": true },
  "rules": { "typescript/prefer-promise-reject-errors": "error" }
}

**Comment by hesprs** [comment]
Strange, I tested to add `--type-aware` to my repro command (`oxlint-tsgolint` installed), but the false positive persists.

**Comment by camc314** [comment]
> Strange, I tested to add `--type-aware` to my repro command (`oxlint-tsgolint` installed), but the false positive persists.

can you share the error? just adding type-aware won't disable the eslint varient of this rule

**Comment by hesprs** [comment]
Oh, my bad. There's a TypeScript equivalence rule `typescript/prefer-promise-reject-errors`. I was still using the non-TypeScript rule.

**Comment by camc314** [comment]
> Oh, my bad. There's a TypeScript equivalence rule `typescript/prefer-promise-reject-errors`. I was still using the non-TypeScript rule.

no worries! - we still don't align with eslint here, i've pushed a change to fix this!
